packages/flutter/lib/src/material/flexible_space_bar.dart

```diff
@@ -108,6 +124,16 @@ class FlexibleSpaceBar extends StatefulWidget {
   /// Defaults to [CollapseMode.parallax].
   final CollapseMode collapseMode;
 
+  /// When [collapseMode] is [CollapseMode.nonePin], boundary height of folding
+  /// effect switching.
+  ///
+  /// When [collapseMode] is [CollapseMode.nonePin], and `currentExtent` is greater
+  /// than [FlexibleSpaceBar.nonePinBoundary], the background widget no collapsing
+  /// effect, when `currentExtent` is less than [FlexibleSpaceBar.nonePinBoundary],
+  /// pin in place until it reaches the min extent. If [FlexibleSpaceBar.nonePinBoundary]
+  /// is 0, the effect is the same as [CollapseMode.none]
+  final double nonePinBoundary;
```

Comment:
Would it be better to allow for configuring pin behavior, rather than adding this nonePin behavior?

Reply:
I think it is not a good choice to use [none] directly, everyone is used to the characteristics of [none]

Reply:
@CyJaySong now we have backgroundBuilder, then why still do we need nonePinBoundary?
Can’t combining none and backgroundBuilder replace `nonePin` and `nonePinBoundary`?

Reply:
@SuhwanCha `nonePin` and `nonePinBoundary` can achieve [CollapseMode.none] and [CollapseMode.pin] two effects appear in the process of collapsing, which is applicable in many scenarios, such as e-commerce projects, like the video I uploaded.

Reply:
@SuhwanCha  In my understanding, `nonePin` and `nonePinBoundary` is a new effect, `backgroundBuilder` is another new feature